src/Css/Style.php

```diff
@@ -1446,6 +1493,16 @@ public function set_prop(string $prop, $val, bool $important = false, bool $clea
                 $val = self::$_defaults[$prop];
             }
 
+            // Do not resolve CSS custom property values when setting them,
+            // because the required custom property might not have been set yet,
+            // or need to be derived from a parent. Let the compute() method
+            // resolve this when the value is needed. Just set the property now.
+            if (is_string($val) && strpos($val, 'var(') !== false) {
+                $this->_props[$prop] = $val;
+                self::$_dependent_props[$prop] = true;
```

Comment:
Since the `var` dependencies are style-specific, using the static map here does not work correctly.

Reply:
Ah, I thought the static would live within each style. I did not realise it would apply to all of them. Reverted.